lib/puppet/parser/scope.rb

```diff
@@ -579,47 +580,51 @@ def inherited_scope
   #
   # @return [Puppet::Parser::Scope] The scope or nil if there is no enclosing scope
   def enclosing_scope
-    if has_enclosing_scope?
-      if parent.is_topscope? or parent.is_nodescope?
+     if has_enclosing_scope?
+      if parent.is_topscope? || parent.is_nodescope?
         parent
       else
         parent.enclosing_scope
       end
-    else
-      nil
-    end
+     end
   end
 
   def is_classscope?
-    resource and resource.type == "Class"
+    resource && resource.type == TYPENAME_CLASS
   end
 
   def is_nodescope?
-    resource and resource.type == "Node"
+    resource && resource.type == TYPENAME_NODE
   end
 
   def is_topscope?
-    @compiler && equal?(@compiler.topscope)
+    equal?(@compiler.topscope)
   end
 
-  def lookup_qualified_variable(class_name, variable_name, position)
-    begin
-      if lookup_as_local_name?(class_name, variable_name)
-        if is_topscope?
-          # This is the case where $::x is looked up from within the topscope itself, or from a local scope
-          # parented at the top scope. In this case, the lookup must ignore local and ephemeral scopes.
-          #
-          handle_not_found(class_name, variable_name, position) unless @symtable.include?(variable_name)
-          @symtable[variable_name]
-        else
-          self[variable_name]
+  # @api private
+  def lookup_qualified_variable(fqn, options)
```

Comment:
This change isn't backward compatible. The method used to be public and the signature changes. Consider keeping the old method with a deprecation warning and name the new method something else.